backend/local/local.go

```diff
@@ -850,6 +944,27 @@ func (o *Object) Update(in io.Reader, src fs.ObjectInfo, options ...fs.OpenOptio
 	if err == nil {
 		err = closeErr
 	}
+
+	if o.translatedLink {
+		if err == nil {
+			// Remove any current symlink or file, if one exsits
+			if _, err := os.Lstat(o.path); err == nil {
+				if removeErr := os.Remove(o.path); removeErr != nil {
+					fs.Errorf(o, "Failed to remove previous file: %v", removeErr)
+					return removeErr
+				}
+			}
+			// Use the contents for the copied object to create a symlink
+			nwc, _ := out.(nopWriterCloser)
```

Comment:
And you can replace these two lines with

    err = os.Symlink(symlinkData.String(), o.path)

And your `nopWriterCloser` becomes redundant (clever though it is!)